client-app/src/desktop/common/SampleTreeWithCheckBoxGrid.js

```diff
@@ -0,0 +1,228 @@
+                align: 'right',
+                width: 130,
+                agOptions: {
+                    aggFunc: 'sum'
+                },
+                renderer: numberRenderer({
+                    precision: 0,
+                    ledger: true,
+                    colorSpec: true,
+                    tooltip: true
+                })
+            },
+            {...emptyFlexCol}
+        ]
+    });
+
+    constructor(props) {
+        super(props);
+        this.loadAsync();
+    }
+
+    render() {
+        const {model} = this,
+            {store} = model;
+
+        return panel({
+            className: this.getClassName(),
+            ...this.getLayoutProps(),
+            item: grid({className: 'sample-tree-checkbox-grid', model}),
+            mask: mask({spinner: true, model: this.loadModel}),
+            bbar: toolbar({
+                omit: this.props.omitToolbar,
+                items: [
+                    storeFilterField({
+                        store,
+                        fields: ['name']
+                    }),
+                    storeCountLabel({
+                        store,
+                        units: 'companies'
+                    }),
+                    filler(),
+                    box('Compact mode:'),
+                    switchInput({
+                        field: 'compact',
+                        model
+                    }),
+                    toolbarSep(),
+                    colChooserButton({gridModel: model}),
+                    exportButton({model, exportType: 'excel'}),
+                    refreshButton({model: this})
+                ]
+            })
+        });
+    }
+
+    //------------------------
+    // Implementation
+    //------------------------
+    loadAsync() {
+        wait(250)
+            .then(() => this.model.loadData(this.sampleData()))
+            .linkTo(this.loadModel);
+    }
+
+    toggleNode(rec) {
+        rec.indeterminate = false;
+        rec.enabled = !rec.enabled;
+        this.setChildren(rec, rec.enabled);
+        this.updateParents(rec);
+        this.model.store.noteDataUpdated();
+    }
+
+    setChildren(rec, enabled) {
+        rec.children.forEach(it => {
+            it.indeterminate = false;
+            it.enabled = enabled;
+            this.setChildren(it, enabled);
+        });
+    }
+
+    updateParents(rec) {
+        if (!rec.parent) return;
+
+        const parent = rec.parent,
+            isAllEnabled = (rec) => rec.children.every(it => it.enabled && isAllEnabled(it)),
+            isAllDisabled = (rec) => rec.children.every(it => !it.enabled && isAllDisabled(it)),
+            allEnabled = isAllEnabled(parent),
+            allDisabled = isAllDisabled(parent),
+            indeterminate = !allEnabled && !allDisabled;
+
+        rec.parent.indeterminate = indeterminate;
+        rec.parent.enabled = allEnabled;
+
+        this.updateParents(rec.parent);
+    }
+
+    sampleData() {
+        return [
```

Comment:
Can we put this data in a shared file for sharing between the grids?  Maybe we will enhance it and make it more realistic.

Reply:
sure, will do.

Reply:
This is done.